deps/compiler-rt.mk

```diff
@@ -0,0 +1,144 @@
+else
+# The standalone compiler-rt build is based on
+# https://github.com/ReservedField/arm-compiler-rt
+CRT_SRCDIR := $(COMPILER_RT_SRCDIR)/lib/builtins
+CRT_ARCH_SRCDIR := $(CRT_SRCDIR)/$(CRT_ARCH)
+CRT_INCLUDES := -I$(CRT_SRCDIR) -I$(CRT_ARCH_SRCDIR)
+CRT_CFLAGS := \
+		-std=gnu99 -fPIC -fno-builtin -fvisibility=hidden \
+		-ffreestanding $(CRT_INCLUDES)\
+		$(if $(USE_CLANG), -Wno-unknown-attributes -Wno-macro-redefined)
+
+##
+# Blacklist a few files we don't want to deal with
+##
+CRT_BLACKLIST := atomic.c atomic_flag_clear.c atomic_flag_clear_explicit.c \
+	atomic_flag_test_and_set.c atomic_flag_test_and_set_explicit.c \
+	atomic_signal_fence.c atomic_thread_fence.c
+
+# TODO(vchuravy) discover architecture flags
+# Discover all files...
+CRT_CFILES := $(wildcard $(CRT_SRCDIR)/*.c)
+CRT_GENERAL_OBJS1 := $(filter-out $(CRT_BLACKLIST:.c=.o), $(notdir $(CRT_CFILES:.c=.o)))
+
+CRT_ARCH_CFILES := $(wildcard $(CRT_ARCH_SRCDIR)/*.c)
+CRT_ARCH_SFILES := $(wildcard $(CRT_ARCH_SRCDIR)/*.S)
+CRT_ARCH_OBJS   := $(notdir $(join $(CRT_ARCH_CFILES:.c=.o),$(CRT_ARCH_SFILES:.S=.o)))
+
+CRT_GENERAL_OBJS := $(filter-out $(CRT_ARCH_OBJS), $(CRT_GENERAL_OBJS1))
+
+CRT_OBJS := $(addprefix $(COMPILER_RT_BUILDDIR)/,$(CRT_GENERAL_OBJS)) \
+	$(addprefix $(COMPILER_RT_BUILDDIR)/$(CRT_ARCH)/,$(CRT_ARCH_OBJS))
+$(info $(CRT_OBJS))
+
+CRT_BUILDDIR := $(COMPILER_RT_BUILDDIR)
+$(CRT_BUILDDIR)/$(CRT_ARCH)/build-configured: $(CRT_BUILDDIR)/build-configured
+	mkdir -p $(dir $@)
+	echo 1 > $@
+
+$(CRT_BUILDDIR)/%.o: $(CRT_SRCDIR)/%.c | $(CRT_BUILDDIR)/build-configured
+	$(CC) $(CRT_CFLAGS) -c $< -o $@
+
+$(CRT_BUILDDIR)/%.o: $(CRT_SRCDIR)/%.S | $(CRT_BUILDDIR)/build-configured
+	$(CC) $(CRT_CFLAGS) -c $< -o $@
+
+$(CRT_BUILDDIR)/$(CRT_ARCH)/%.o: $(CRT_ARCH_SRCDIR)/%.c | $(CRT_BUILDDIR)/$(CRT_ARCH)/build-configured
+	$(CC) $(CRT_CFLAGS) -c $< -o $@
+
+$(CRT_BUILDDIR)/$(CRT_ARCH)/%.o: $(CRT_ARCH_SRCDIR)/%.S | $(CRT_BUILDDIR)/$(CRT_ARCH)/build-configured
+	$(CC) $(CRT_CFLAGS) -c $< -o $@
+
+$(COMPILER_RT_BUILDDIR)/$(COMPILER_RT_LIBFILE): $(CRT_OBJS)
+	$(CC) -shared -o $@ $^
+
+endif
+
+$(COMPILER_RT_SRCDIR)/source-extracted: | $(COMPILER_RT_TAR)
+ifneq ($(COMPILER_RT_TAR),)
+	$(JLCHECKSUM) $(COMPILER_RT_TAR)
+endif
+	mkdir -p $(COMPILER_RT_SRCDIR)
+	$(TAR) -C $(COMPILER_RT_SRCDIR) --strip-components 1 -xf $(COMPILER_RT_TAR)
+	echo 1 > $@
+
+$(COMPILER_RT_BUILDDIR)/build-configured:
+	mkdir -p $(dir $@)
+	echo 1 > $@
+
+get-compiler-rt: $(COMPILER_RT_TAR)
+ifeq ($(STANDALONE_COMPILER_RT), 0)
+extract-compiler-rt: #NONE
+else
+extract-compiler-rt: $(COMPILER_RT_SRCDIR)/source-extracted
+endif
+check-compiler-rt: #NONE
+fast-check-compiler-rt: #NONE
+configure-compiler-rt: $(COMPILER_RT_BUILDDIR)/build-configured
+clean-compiler-rt:
+	rm -rf $(COMPILER_RT_BUILDDIR)
+compile-compiler-rt: $(COMPILER_RT_BUILDDIR)/$(COMPILER_RT_LIBFILE)
+install-compiler-rt: $(COMPILER_RT_BUILDDIR)/$(COMPILER_RT_LIBFILE)
+	cp $(COMPILER_RT_BUILDDIR)/$(COMPILER_RT_LIBFILE) $(build_private_libdir)/
```

Comment:
@vtjnash When I discussed this with you, you mentioned that I might not need to use the macro since I only install two files?
I am also happy to use the staged-install mechanism, but I hadn't had time to understand it yet.